doc/src/Intro.rst

```diff
@@ -76,6 +90,12 @@ For dependencies on other boost libraries, they can be listed with the ``DEPENDS
             config
     )
 
+.. This is interesting. I like the reduction in boilerplate, but I question the
+   choice to go in such a different direction from the standard CMake way of
+   doing things. It strikes me as a possibly leaky abstraction where folks will
+   have a mix of code which uses both the CMake way and the BCM way; maintainers
+   will have to be familiar with both.
```

Comment:
Definitely agree with camio on this.  See this Youtube video by Daniel Pfeiffer

https://www.youtube.com/watch?v=bsXLMQ6WgIk

Having a set of guidelines, templates, and conventions on how to use modern CMake (3.0+) is a better way to go than reinventing CMake's conventions.